src/codec.rs

```diff
@@ -782,36 +782,71 @@ impl<T: Encode> Encode for [T] {
 	}
 }
 
-/// Read an `u8` vector from the given input.
-pub(crate) fn read_vec_u8<I: Input>(input: &mut I, len: usize) -> Result<Vec<u8>, Error> {
+/// Create a `Vec<T>` by casting directly from a buffer of read `u8`s
+///
+/// `T` must encodes to its byte representation, size of `T` must be less or equal to
```

Comment:
> `T` must encodes to its byte representation

What you mean by this?

Reply:
I was thinking the binary representation of the type, but I changed to https://github.com/paritytech/parity-scale-codec/pull/231/files#r546887679

But this makes me think, when encoding/decoding `Vec<T>` we don't take care of the endian of the plateform. `Encode` calls `as_byte_slice` which has different behavior depending on endian of the plateform. Isn't this wrong ?

Reply:
I guess we should do this optimisation only if `cfg(target_endian = little)`

Reply:
> When encoding/decoding `Vec<T>` we don't take care of the endian of the plateform. `Encode` calls `as_byte_slice` which has different behavior depending on endian of the plateform. Isn't this wrong ?

That is a very good point :see_no_evil:

Reply:
I made a commit to fix this